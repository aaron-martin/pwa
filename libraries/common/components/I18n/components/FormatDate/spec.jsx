/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import I18n from '../../index';

describe('<FormatDate />', () => {
  let renderedElement;
  const testLocales = {
    greeting: 'Hello {date}',
  };
  const timestamp = 123456789000;
  const formattedDate = 'Nov 29, 1973';
  const format = 'medium';
  const langCode = 'en-US';

  /**
   * Renders the component.
   * @param {Object} props The component props.
   */
  const renderComponent = (props) => {
    renderedElement = mount(
      <I18n.Provider {...props}>
        <div>
          <span className="only-date">
            <I18n.Date timestamp={timestamp} format={format} />
          </span>
          <span className="text-with-date">
            <I18n.Text string="greeting">
              <I18n.Date forKey="date" timestamp={timestamp} format={format} />
            </I18n.Text>
          </span>
        </div>
      </I18n.Provider>
    );
  };

  beforeEach(() => {
    renderComponent({
      lang: langCode,
      locales: testLocales,
    });
  });

  describe('Given the component was mounted to the DOM', () => {
    it('should match snapshot', () => {
      expect(renderedElement).toMatchSnapshot();
    });

    it('should render formatted date', () => {
      const text = renderedElement.find('.only-date').text();
      expect(text).toBe(formattedDate);
    });

    it('should render within translated text', () => {
      const text = renderedElement.find('.text-with-date').text();
      expect(text).toBe(`Hello ${formattedDate}`);
    });
  });
});
