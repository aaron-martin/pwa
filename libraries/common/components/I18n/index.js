/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import I18nProvider from './components/I18nProvider';
import Placeholder from './components/Placeholder';
import Translate from './components/Translate';
import FormatDate from './components/FormatDate';
import FormatTime from './components/FormatTime';
import FormatPrice from './components/FormatPrice';

export default {
  Provider: I18nProvider,
  Text: Translate,
  Date: FormatDate,
  Time: FormatTime,
  Placeholder,
  Price: FormatPrice,
};
