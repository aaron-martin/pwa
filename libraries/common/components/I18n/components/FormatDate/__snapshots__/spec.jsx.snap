// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatDate /> Given the component was mounted to the DOM should match snapshot 1`] = `
<I18nProvider
  lang="en-US"
  locales={
    Object {
      "greeting": "Hello {date}",
    }
  }
>
  <div>
    <span
      className="only-date"
    >
      <FormatDate
        format="medium"
        timestamp={123456789000}
      >
        <span>
          Nov 29, 1973
        </span>
      </FormatDate>
    </span>
    <span
      className="text-with-date"
    >
      <Translate
        className=""
        params={Object {}}
        string="greeting"
      >
        <span
          className=""
        >
          Hello 
          <FormatDate
            forKey="date"
            format="medium"
            timestamp={123456789000}
          >
            <span>
              Nov 29, 1973
            </span>
          </FormatDate>
        </span>
      </Translate>
    </span>
  </div>
</I18nProvider>
`;
